collector/receiver/otelarrowreceiver/internal/arrow/arrow.go

```diff
@@ -374,16 +374,29 @@ func (r *Receiver) anyStream(serverStream anyStreamServer, method string) (retEr
 
 func (r *Receiver) processAndConsume(ctx context.Context, method string, arrowConsumer arrowRecord.ConsumerAPI, req *arrowpb.BatchArrowRecords, serverStream anyStreamServer, authErr error) error {
 	var err error
+	var bytesProcessed int64
 
 	ctx, span := r.tracer.Start(ctx, "otel_arrow_stream_recv")
 	defer span.End()
 
+	var err error
+	defer func() {
+		// Due to potential double compression the netstats code knows uncompressed bytes
+		// value can be unreliable. Add span attributes for uncompressed size and set
+		// span Status if an error is returned.
+		var sized netstats.SizesStruct
+		sized.Method = method
+		sized.Length = bytesProcessed
+		s.netReporter.SetSpanSizeAttributes(ctx, sized)
+		s.netReporter.SetSpanError(ctx, err)
```

Comment:
Similar statements as the exporter side -- I think the SetSpanSizeAttributes can be moved to nearby the location where `r.netReporter.CountReceive` is called with the uncompressed size.

The span status can be left here, but directly, not via a netstats API.

Reply:
Fixed